tests/dialects/test_starrocks.py

```diff
@@ -5,6 +5,61 @@
 class TestStarrocks(Validator):
     dialect = "starrocks"
 
+    def test_ddl(self):
+        # Test the different wider DECIMAL types
+        self.validate_identity(
+            "CREATE TABLE foo (col0 DECIMAL(9, 1), col1 DECIMAL32(9, 1), col2 DECIMAL64(18, 10), col3 DECIMAL128(38, 10)) DISTRIBUTED BY HASH (col1) BUCKETS 1"
+        )
+        # Test the PRIMARY KEY with ENGINE property
+        self.validate_identity(
+            "CREATE TABLE foo2 (col1 BIGINT) ENGINE=OLAP PRIMARY KEY (col1) DISTRIBUTED BY HASH (col1) BUCKETS 1"
+        )
```

Comment:
Nit, I'm in favor of reducing noise by condensing similar test cases through loops; For example, you can have a for-loop for the different partition schemes, for the existence (or not) of engine/buckets and so on. 

This helps both extending old cases and writing new ones because it's much easier to spot which keywords are tested instead of skimming through individual identity checks.